### PR TITLE
TST: Refactor test_expressions.py

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

Enable more `pytest.mark.parametrize`

### CODE REVIEW

File: pandas/tests/test_expressions.py [hunk 1]
```diff
@@ -134,8 +115,31 @@ def run_frame(self, df, other, flex: bool):
         ],
     )
     @pytest.mark.parametrize("flex", [True, False])
-    def test_arithmetic(self, df, flex):
-        self.run_frame(df, df, flex)
+    @pytest.mark.parametrize("arith", ["gt", "lt", "ge", "le", "eq", "ne"])
```
Comment (jreback) [inline pandas/tests/test_expressions.py:118]: we have a fixture for this

**Review by jreback — APPROVED** [review]
cc @jbrockmendel if any comments

File: pandas/tests/test_expressions.py [hunk 2]
```diff
@@ -344,18 +348,18 @@ def test_bool_ops_column_name_dtype(self, test_input, expected):
         result = test_input.loc[:, ["a", "dtype"]].ne(test_input.loc[:, ["a", "dtype"]])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("df", [_frame])
```
Comment (jbrockmendel) [inline pandas/tests/test_expressions.py:351]: just define `df = _frame'` inside?

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Pull request merged**